composer/devices/device_gpu.py

```diff
@@ -42,6 +44,9 @@ def __init__(
     ):
         if not torch.cuda.is_available():
             raise ValueError('DeviceGPU cannot be created as torch.cuda is not available.')
+        if torch_dist.is_gloo_available() or version.parse(torch.__version__) >= version.parse('2.3.0.dev'):
+            # Composer checkpoint load / save from before torch 2.3.0 onwards is not compatible with gloo + nccl backends.
+            DeviceGPU.dist_backend = 'cuda:nccl,cpu:gloo'
```

Comment:
Let's just pin to >= 2.3.1 for simplicity, I don't really want to muck w dev branches

Reply:
Changed to `2.3.0` -- will only not be applied for torch nightly 2.3.0 (2.3.0dev).

It works for 2.3.0 and since a lot of ppl are still using 2.3.0 images this is a nice improvement to have